packages/critters/src/dom.js

```diff
@@ -16,9 +16,37 @@
 
 import { selectAll, selectOne } from 'css-select';
 import { parseDocument, DomUtils } from 'htmlparser2';
+import { parse as selectorParser } from 'css-what';
 import { Element, Text } from 'domhandler';
 import render from 'dom-serializer';
 
+let classCache = null;
+let idCache = null;
+
+function buildCache(container) {
```

Comment:
Both caches `idCache` and `classCache` completely disregard the document/contain. Wouldn't this cause incorrect result, like false positive matches when running critters on multiple documents?

Reply:
Could you explain this a bit more, not sure what disregards document/contain means.

Reply:
Just want to explain how this works currently -
`buildCache` collects all classnames and ids that occur inside the container (the container marked for critical css).
When evaluating critical CSS we check if the selector is a single class selector (eg: `.container`) and then check for a cache hit. Otherwise we run the default selector matcher.

We make sure `html` and `body` is a match always and then evaluate other CSS based on the critters container. Also the cache is reset for every new document.

Reply:
Oh right, I missed the part that the cache is reset for every document.